radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/steady_state/deterministic/full_function/ApiTest.java

```diff
@@ -62,63 +62,78 @@
+          new ActorConfiguration(BalanceReconciler::new, 1, 10),
+          new ActorConfiguration(RandomNodeRestarter::new, 1, 10),
+          new ActorConfiguration(NativeTokenRewardsChecker::new, 1, 100),
+          new ActorConfiguration(ApiBalanceToRadixEngineChecker::new, 1, 200));
 
-  @BeforeClass
-  public static void startRadixTest() throws Exception {
-    TestSetupUtils.installBouncyCastleProvider();
+  public ApiTest(Module forkModule, Module byzantineModule) {
+    super(forkModule, byzantineModule);
+    this.setActorConfigurations(ACTOR_CONFIGURATIONS);
+  }
 
-    serialization = DefaultSerialization.getInstance();
+  // The following class is created as a workaround as gradle cannot run the tests inside a test
+  // class in parallel. We can achieve some level of parallelism splitting the tests across
+  // different test classes.
 
-    JSONObject runtimeConfigurationJSON = new JSONObject();
-    if (Radix.class.getResourceAsStream("/runtime_options.json") != null) {
-      runtimeConfigurationJSON =
-          new JSONObject(
-              IOUtils.toString(Radix.class.getResourceAsStream("/runtime_options.json")));
+  public static class ApiTest2 extends ApiTest {
+    public ApiTest2() {
+      super(new RadixEngineForksLatestOnlyModule(config.overrideMaxRounds(98)), null);
     }
+  }
 
-    properties = new RuntimeProperties(runtimeConfigurationJSON, null);
-
-    // Tests need this
-    properties.set("network.host_ip", "127.0.0.1");
-
-    if (dbLocation == null) {
-      dbLocation = folder.getRoot().getAbsolutePath();
+  public static class ApiTest1 extends ApiTest {
+    public ApiTest1() {
+      super(new RadixEngineForksLatestOnlyModule(config), null);
     }
-    properties.set("db.location", dbLocation);
   }
 
-  @AfterClass
-  public static void stopRadixTest() {
-    serialization = null;
-    dbLocation = null;
-    properties = null;
+  public static class ApiTest0 extends ApiTest {
+    public ApiTest0() {
+      super(
+          new ForkOverwritesWithShorterEpochsModule(config),
+          new ForkOverwritesWithShorterEpochsModule(config.removeSigsPerRoundLimit()));
+    }
   }
 
-  protected Serialization getSerialization() {
-    return Objects.requireNonNull(serialization, "serialization was not initialized");
+  public static class ApiTest3 extends ApiTest {
+    public ApiTest3() {
+      super(
+          new RadixEngineForksLatestOnlyModule(
+              config
+                  .overrideMaxRounds(98)
+                  .overrideFeeTable(FeeTable.create(PER_BYTE_FEE, Map.of()))),
+          null);
+    }
   }
 
-  protected RuntimeProperties getProperties() {
-    return Objects.requireNonNull(properties, "properties was not initialized");
+  public static class ApiTest4 extends ApiTest {
+    public ApiTest4() {
+      super(
+          new ForkOverwritesWithShorterEpochsModule(
+              config.overrideFeeTable(FeeTable.create(PER_BYTE_FEE, Map.of()))),
+          null);
+    }
   }
```

Comment:
Didi you see benefits with this number of parallel tests?

Reply:
On my local machine yes